docs/docs/validators/partial-set-security-for-validators.md

```diff
@@ -0,0 +1,118 @@
+---
+sidebar_position: 6
+---
+
+# Partial Set Security
+[Partial Set Security](../features/partial-set-security.md) allows consumer chain to join as Opt-In or Top N.
+Here, we show how a validator can opt in, opt out, or set a custom commission rate on a consumer chain, as well
+as useful queries that a validator can use to figure out which chains it has to validate, etc.
+
+## Messages
+### How to opt in to a consumer chain?
+
+:::warning
+A validator is automatically opted in to a Top N chain if the validator belongs to the top N% of the validators on the provider chain.
+:::
+
+In a Top N chain, a validator that does not belong to the top N% of the validators on the provider can still choose
+to opt in to a consumer chain. In other words, validators can opt in, in both Opt-In and Top N chains.
+
+A validator can opt in to a consumer chain by issuing the following message:
+```bash
+interchain-security-pd tx provider opt-in <consumer-chain-id> <optional consumer-pub-key>
+```
+
+where
+- `consumer-chain-id` is the string identifier of the consumer chain the validator wants to opt in to;
+- `consumer-pub-key` is an **optional** field that corresponds to the public key the validator wants to use on the
+consumer chain, and it has the following format `{"@type":"/cosmos.crypto.ed25519.PubKey","key":"<key>"}`.
+
+A validator can opt in to an existing consumer chain that is already running, or to a [proposed](../features/proposals.md)
+consumer chain that is still being voted on. A validator can use the following command to retrieve the currently existing
+consumer chains:
+```bash
+interchain-security-pd query provider list-consumer-chains
+```
+and this command to see the currently proposed consumer chains:
+```bash
+interchain-security-pd query provider list-proposed-consumer-chains
+```
+
+
+:::tip
+By setting the `consumer-pub-key`, a validator can both opt in to a chain, as well as, assign a
+public key on a consumer chain. Note that a validator can always issue a [key assignment](../features/key-assignment.md)
+at a later stage to assign a new consumer public key on a chain. The key-assignment [rules](../features/key-assignment.md#rules)
+still apply when setting `consumer-pub-key` when opting in.
+:::
+
+:::info
+A validator is only eligible for consumer rewards from a consumer chain if the validator is opted in on that chain.
+:::
+
+### How to opt out from a consumer chain?
+A validator can opt out from a consumer by issuing the following message:
+
+```bash
+interchain-security-pd tx provider opt-out <consumer-chain-id>
+```
+where
+- `consumer-chain-id` is the string identifier of the consumer chain.
+
+:::warning
+A validator cannot opt out from a Top N chain if it belongs to the top N% validators on that chain.
```

Comment:
What does *that chain* mean here? It's the top N% of the provider, right? I'd just say
```suggestion
A validator cannot opt out from a Top N chain if it belongs to the top N% validators of the provider.
```